src/controller/WebController.js

```diff
@@ -0,0 +1,173 @@
+      this.#lottoArray.push(lotto);
+    }
+  };
+
+  printLotto = (lottoAmount) => {
+    $(".purchase-amount").innerHTML = `총 ${lottoAmount}개를 구매하였습니다.`;
+
+    const lottoList = this.#lottoArray
+      .map((lotto) => {
+        lotto.sortLottoNumbers();
+        return `<li>🎟️ ${lotto.getLottoNumbers()}</li>`;
+      })
+      .join("");
+
+    $(".print-lottos-list").innerHTML = lottoList;
+  };
+
+  getWinningNumbers = () => {
+    try {
+      for (let i = 0; i < SETTINGS.MAX_WINNING_NUMBER_LENGTH; i++) {
+        if ($$(".winning-number")[i].value === "") {
+          throw new Error(ERROR_MESSAGE.EMPTY_INPUT);
+        }
+        this.#winningLotto.push(parseInt($$(".winning-number")[i].value, 10));
+      }
+
+      this.validateWinningNumbers(this.#winningLotto);
+      return this.#winningLotto;
+    } catch (e) {
+      alert(e.message);
+      this.#winningLotto = [];
+    }
+  };
+
+  validateWinningNumbers = (winningLottoNumbers) => {
+    Validator.isDuplicatedNumber(winningLottoNumbers);
+    Validator.isCorrectLength(winningLottoNumbers);
+    
+    for (let i = 0; i < winningLottoNumbers.length; i++) {
+      this.checkEachNumber(winningLottoNumbers[i]);
+    }
+  };
+
+  checkEachNumber = (eachNumber) => {
+    Validator.isNumber(eachNumber);
+    Validator.isCorrectRange(eachNumber);
+    Validator.isPositiveInteger(eachNumber);
+  };
+
+  getBonusNumber = () => {
+    try {
+      this.#bonusNumber = parseInt($(".bonus-number").value, 10);
+      this.validateBonusNumber();
+      this.checkEachNumber(this.#bonusNumber);
+      return this.#bonusNumber;
+    } catch (e) {
+      alert(e.message);
+      this.#bonusNumber = 0;
+    }
+  };
+
+  validateBonusNumber = () => {
+    Validator.hasBonusNumber(this.#bonusNumber, this.#winningLotto);
+  };
+
+  compareLottos = (e) => {
+    e.preventDefault();
+
+    this.getWinningNumbers();
+    this.getBonusNumber();
+    const lottos = new Lottos(this.#lottoArray);
+
+    lottos.getLottos().forEach((lotto) => {
+      lotto.compareNumbers(this.#winningLotto);
+      lotto.checkBonusNumber(this.#bonusNumber);
+    });
+
+    lottos.compareLottosScore();
+    this.printResult(lottos);
+  };
+
+  printResult = (lottos) => {
+    $(".lotto-result-wrap").classList.add("show");
+    $(".three").innerHTML = `${lottos.getLottoRanking()[SCORE.THREE]}개`;
+    $(".four").innerHTML = `${lottos.getLottoRanking()[SCORE.FOUR]}개`;
+    $(".five").innerHTML = `${lottos.getLottoRanking()[SCORE.FIVE]}개`;
+    $(".five-bonus").innerHTML = `${lottos.getLottoRanking()[SCORE.FIVE_BONUS]}개`;
+    $(".six").innerHTML = `${lottos.getLottoRanking()[SCORE.SIX]}개`;
+
+    lottos.calculateBenefit();
+    const totalBenefit = lottos.getBenefitRate($(".input-money").value);
+    $(".result-message").innerHTML = `당신의 총 수익률은 ${totalBenefit}%입니다.`;
+    
+    this.resetScore(lottos);
+    this.#winningLotto = [];
+  };
+
+  resetScore = (lottos) => {
+    for (const lotto of lottos.getLottos()) {
```

Comment:
많은 반복 구문 중 `for of`를 사용하신 이유가 있을가요?

Reply:
lottos는 발행된 로또 객체들을 가지고 있는 객체입니다. 각 로또 객체가 가지고 있는 메서드 중 resetScore()를 호출하기 위해서 배열 값을 순환하는 `for of`를 사용하였습니다!

Reply:
```suggestion
    lottos.getLottos().forEach((lotto) => {
      lotto.resetScore()
    })
```

`for of`를 쓴 특별한 이유는 없었습니다. forEach를 사용하니 좀 더 간결해질 수 있을 것 같습니다.